barista/src/main/java/com/markelliot/barista/Server.java

```diff
@@ -62,7 +68,15 @@ private void start() {
      * included automatically.
      */
     public void stop() {
-        undertow.stop();
+        shutdownHandler.shutdown();
+        shutdownHandler.addShutdownListener(shutdownSuccessful -> undertow.stop());
```

Comment:
I think we ought to do this before calling shutdown to be sure that the listener is definitely called

Reply:
you can only add shutdown hooks after you've told the handler to shutdown.

https://github.com/undertow-io/undertow/blob/master/core/src/main/java/io/undertow/server/handlers/GracefulShutdownHandler.java#L169-L171

Reply:
Oh that’s weird